files-csharp/ComparingTwoFiles/ComparingTwoFiles/FileComparer.cs

```diff
@@ -21,23 +21,54 @@ public static bool CompareByNameAndSize(string file1, string file2)
 
     public static bool CompareByBytes(string file1, string file2)
     {
-        var bytes1 = File.ReadAllBytes(file1);
-        var bytes2 = File.ReadAllBytes(file2);
+        const int bufferSize = 1024;
+        using var stream1 = new FileStream(file1, FileMode.Open, FileAccess.Read);
+        using var stream2 = new FileStream(file2, FileMode.Open, FileAccess.Read);
 
-        if (bytes1.Length != bytes2.Length)
+        if (stream1.Length != stream2.Length)
         {
             return false;
         }
 
-        return Enumerable.SequenceEqual(bytes1, bytes2);
+        Span<byte> buffer1 = new byte[bufferSize];
+        Span<byte> buffer2 = new byte[bufferSize];
+
+        while (true)
+        {
+            var bytesRead1 = stream1.Read(buffer1);
+            var bytesRead2 = stream2.Read(buffer2);
+
+            if (bytesRead1 == bytesRead2 && bytesRead1 == 0)
+            {
+                return true;
+            }
+
+            if (!MemoryExtensions.SequenceEqual(buffer1, buffer2))
```

Comment:
Good switch to the stream and spans. Just one cosmetic change here and in the next method. Since `SequenceEqual()` is an extension method, this can be simplified to:
`buffer1.SequenceEqual(buffer2)`